src/communication/stream_transfer_sender.cpp

```diff
@@ -105,6 +122,24 @@ namespace cubstream
 		break;
 	      }
 	  }
+
+	if (this_producer_channel.m_last_sent_position < this_producer_channel.m_stream.get_last_committed_pos ())
+	  {
+	    /* send all stream data before termination */
+	    return;
+	  }
+
+	if (this_producer_channel.is_termination_phase ())
```

Comment:
Would be nice to have only "send" in transfer_sender_task. The "recv" calls must be called in other threads at begin/end.
Similarly for transfer_receiver_task (only "recv" calls - move "send" calls into another thread).
Maybe done in a separate jira issue.

Reply:
I created a new issue : CBRD-23127